qiskit_nature/transformers/particle_hole_transformer.py

```diff
@@ -28,5 +45,2102 @@ def transform(self, q_molecule: QMolecule) -> QMolecule:
+
+        h1_old_matrix = h1_old_matrix.copy()
+
+        # put labels of occupied orbitals in the list in interleaved spin convention
+        n_occupied = []
+        for a_i in range(num_alpha):
+            n_occupied.append(2 * a_i)
+        for b in range(num_beta):
+            n_occupied.append(2 * b + 1)
+
+        for r in range(n_qubits):
+            for s in range(n_qubits):  # pylint: disable=invalid-name
+                for i in n_occupied:
+                    h1_old_matrix[r][s] += h2_old_matrix[r][i][s][i].copy() - \
+                                           h2_old_matrix[r][i][i][s].copy()
+        identities_new_sum = 0
+
+        for i in range(n_qubits):
+            for j in range(n_qubits):
+                indices_1 = [i, j]
+                array_mapping_1 = ['adag', 'a']
+
+                h1_new_matrix = np.zeros([n_qubits, n_qubits])
+                h2_new_matrix = np.zeros([n_qubits, n_qubits, n_qubits, n_qubits])
+
+                h1_new_matrix, h2_new_matrix, identities = self._normal_order_integrals(
+                    n_qubits, n_occupied, indices_1, array_mapping_1, h1_old_matrix,
+                    h2_old_matrix, h1_new_matrix, h2_new_matrix)
+
+                h1_new_sum += h1_new_matrix
+                h2_new_sum += h2_new_matrix
+                identities_new_sum += identities
+
+        for i in range(n_qubits):
+            for j in range(n_qubits):
+                for k in range(n_qubits):
+                    for l_i in range(n_qubits):
+                        array_to_be_ordered = [i, j, k, l_i]
+
+                        array_mapping_2 = ['adag', 'adag', 'a', 'a']
+
+                        h1_new_matrix = np.zeros([n_qubits, n_qubits])
+                        h2_new_matrix = np.zeros([n_qubits, n_qubits, n_qubits, n_qubits])
+
+                        h1_new_matrix, h2_new_matrix, identities = self._normal_order_integrals(
+                            n_qubits, n_occupied, array_to_be_ordered, array_mapping_2,
+                            h1_old_matrix, h2_old_matrix, h1_new_matrix, h2_new_matrix)
+
+                        h1_new_sum += h1_new_matrix
+                        h2_new_sum += h2_new_matrix
+                        identities_new_sum += identities
+
+        h2_new_sum = np.einsum('IKMJ->IJKM', h2_new_sum)
+
+        self._h1 = h1_new_sum
+        self._h2 = h2_new_sum
+
+        self._convert_to_block_spins()
+
+        print('h1',self._h1)
+
+        q_molecule_new = copy.deepcopy(q_molecule)
+
+        q_molecule_new.mo_onee_ints = self._h1[:int(self._h1.shape[0]/2),:int(self._h1.shape[1]/2)]
+
+        # TODO calculation of mo_eri is still wrong, because of the doubling of the space.
+        q_molecule_new.mo_eri_ints = self._h2[:int(self._h2.shape[0]/2),:int(self._h2.shape[1]/2),
+                                              :int(self._h2.shape[2]/2),:int(self._h2.shape[3]/2)]
+
+        q_molecule_new.energy_shift['ParticleHoleTransformer'] = identities_new_sum
```

Comment:
Just noting, since its not in todo list, that the dipole matrices need to be adjusted too I think.

Reply:
I do not think that we ever changed the dipole integrals accordingly in the PH picture. Do you have any reference in the old code for this? I cannot recall any implementation on dipole ints.

Reply:
In the past PH acted on h1/h2 of FermionicOperator. We built out main operator (hamiltonian) and all dipoles as a set of FermionicOperators. They all transformed under the active space reduction, particle hole etc transformation capabilities of the FermionicOperator. You can see it here https://github.com/Qiskit/qiskit-nature/blob/f161ae83de1655aa55dde5dbf4fbbc32e4bb1348/qiskit_nature/transformations/fermionic_transformation.py#L301